discovery-provider/src/model_validator.py

```diff
@@ -0,0 +1,168 @@
+            }
+        }
+    '''
+    models_to_schema_and_fields_dict = {}
+    BASE_PATH = './src/schemas/'
+    FILE_NAME = ''
+
+    # Default field is set to None to validate the entire model
+    @classmethod
+    def validate(cls, to_validate, model, field=None):
+        cls.FILE_NAME = model.lower() + '_schema.json'
+        try:
+            schema = cls.get_schema_for_field(field, model)
+            validator = Draft7Validator(schema)
+
+            found_invalid_field = False
+            errors = []
+            for error in sorted(validator.iter_errors(to_validate), key=str):
+                found_invalid_field = True
+                errors.append(error.message)
+
+            if found_invalid_field:
+                raise ValidationError(f"Instance {to_validate} is not proper. Errors: {errors}")
+
+        except ValidationError as ve:
+            raise ve
+        except SchemaError as se:
+            raise se
+        except BaseException as e:
+            # one of many errors specified in helper methods
+            raise e
+
+    # If field is None, return the entire model schema
+    @classmethod
+    def get_schema_for_field(cls, field, model):
+        try:
+            # If model is not present in dict, init its schema and its field subschemas
+            if model not in cls.models_to_schema_and_fields_dict:
+                schema = cls.init_model_schemas(model)
+                if not schema:
+                    error_msg = f"ModelValidation failed. Are you sure the schema for {model} is present and proper?"
+                    raise RuntimeError(error_msg)
+
+            if field == None:
+                return cls.models_to_schema_and_fields_dict[model]['model_schema']
+
+            # Else, return the specified field schema
+            return cls.models_to_schema_and_fields_dict[model]['field_schema'][field]
+        except BaseException as e:
+            raise e
+
+    @classmethod
+    def init_model_schemas(cls, model):
+        cls.FILE_NAME = model.lower() + '_schema.json'
+        try:
+            # Load in the model schema in /schemas
+            schema = cls.load_schema_from_path(model)
+
+            model_properties = {
+                "model_schema": schema, # schema for the entire model
+                "field_schema": {}, # schema for just a field in model
+                "fields": cls._get_required(schema, model) # list of fields in model
+            }
+
+            cls.models_to_schema_and_fields_dict[model] = model_properties
+
+            # Create all the subschemas for each individual field
+            for field in model_properties['fields']:
+                # Create a deep copy of the entire schema to generate a schema that only
+                # validates one field of the entire model
+                cls.create_field_schema(schema, field, model)
+
+            return schema
+        except BaseException as e:
+            logger.exception(e)
+            return None
```

Comment:
maybe add a comment explaining why we want to absorb this error

Reply:
@vicky-g do you know why this is BaseException as opposed to regular Exception or something else?

Reply:
when i used Exception as the object, the pylint failed and mentioned that Exception was too general. did some digging around and pylint accepts the exception as BaseException. 

BaseException is the super most class for all exceptions https://docs.python.org/3/library/exceptions.html#exception-hierarchy